src/test/java/eu/elixir/ega/ebi/it/res/FileControllerIntgTest.java

```diff
@@ -0,0 +1,109 @@
+package eu.elixir.ega.ebi.it.res;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import org.junit.Test;
+
+import com.google.common.io.CountingInputStream;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+/**
+ * Class to do the integration test for FileController in ega-data-api-key.
+ * 
+ * @author amohan
+ *
+ */
+public class FileControllerIntgTest extends ResBase {
+
+    private final static String FILE_PATH = "/file/archive/";
+    private OkHttpClient client;
+
+    /**
+     * Verify the api call /file/archive/{id} and check status is
+     * {@link org.apache.http.HttpStatus#SC_OK}. Also checks the response body
+     * should not be null.
+     */
+    @Test
+    public void testGetArchiveFile() {
+
+        String url = baseURI + ":" + port + FILE_PATH + fileId + "?destinationFormat=plain";
+        requestRequest = new Request.Builder().url(url).build();
+
+        client = SSLUtilities.getUnsafeOkHttpClient();
+
+        // Set up Digest
+        MessageDigest md = null;
+        try {
+            md = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException ex) {
+        }
+
+        Response response = null;
+        try {
+            response = client.newCall(requestRequest).execute();
+        } catch (IOException ex) {
+            System.out.println("An Error has occurred : " + ex.toString());
+            System.out.println("\t\t" + url);
+            return;

```

Comment:
The same comment for the exception. You capture it and print but then you return so maybe the test then passes if this fails.

Reply:
Added throws exception

Reply:
For the expections now. Do you want to capture it  with the try and catch and rethrow it? Or maybe just removing the try & catch is enough and on any exception the test just will fail.

Reply:
oh so sorry my mistake. I have corrected it.